rfd/0029-account-lifecycle.md

```diff
@@ -0,0 +1,220 @@
+The email transport is not guaranteed to be secure, and email itself can be hacked.
+
+**Getting new tokens**
+
+If any user is logged into Teleport UI, they can regenerate new password recovery tokens:
+
+* Alice goes to her account settings and chooses to generate new set of recovery tokens.
+* Teleport asks for a second factor before generating tokens.
+* A new set of tokens is printed for Alice. The message makes it clear
+that once closed, the tokens are no longer going to be revealed and that any previous
+tokens are invalid.
+
+**Internal request for cancelling the account**
+
+To request account cancellation, any account owner should:
+
+* Be logged into Teleport UI.
+* Press a "cancel my account" button in the UI.
+* UI should verify that they own a second factor.
+* Verify that they have write permission on the billing.
+
+```
+We have marked your account as inactive. In 10 days, we are going to delete the account and all it's data.
+To prevent this, log into Teleport and press "revert account cancelation".
+```
+
+The account enters inactive state:
+
+**Inactive account state**
+
+* Teleport should not disconnect any resources connected to the cloud account during the grace period.
+Otherwise it is possible for attacker overtaking an account owner to cause system wide outage.
+* Account owner will be able to login in the account, but will
+see message of the day displayed in UI and CLI:
+
+```
+Your account was marked as inactive at the request of `email`. In 10 days, we are going to delete the account and all it's data.
+```
+
+* Any user with `billing` write permission can login into the account and is greeted with a warning:
+
+```
+Your account was marked as inactive at the request of `email`.
+To prevent account from being deleted, log into Teleport and press "revert account cancelation".
+```
+
+All other users with billing write permission should receive the email with the same message as above.
+
+Any account owner should be able to revert account cancellation.
+We should ask already logged in account owner for a second factor proof to revert account cancellation to prevent
+a possible account overtake by a malicious actor who wishes to keep the account for accessing
+the resources.
+
+Once system leaves from inactive account state, all account owners
+should receive an email:
+
+```
+Your account was marked as active at the request of `email`.
+```
+
+The possible vectors of attack is for malicious actor to overtake someone's
+account with a request to delete all the data. The grace period and active privileged
+user should mitigate against that. Email sent to other account owners
+should alert other owners about account cancellation.
+
+If there are no other owners, other active users will notice the message of the day in the UI and CLI.
+
+Teleport cloud support should be able to revert account cancelation if they see a zendesk
```

Comment:
I wonder if this truly should be able to revert, or just delay/extend? Otherwise this looks like a reversal based on just email, in which case the process could just send an email link to be equivelant.